test/e2e/network_segmentation.go

```diff
@@ -236,9 +238,15 @@ var _ = Describe("Network Segmentation", func() {
 							)
 						}
 
+						By("ensure enough schedable nodes exist")
+						nodes, err := e2enode.GetBoundedReadySchedulableNodes(context.Background(), cs, 1)
+						Expect(err).NotTo(HaveOccurred())
+						nodeName := nodes.Items[0].Name
+						udnPodConfig.nodeSelector = map[string]string{nodeHostnameKey: nodeName}
+
```

Comment:
as above, you moved the node setup without changing the semantics much, but also removed error checking.

also, you overwrite `udnPodConfig.nodeSelector` (with the same value) below